pkg/optimization/substat_details.go

```diff
@@ -0,0 +1,551 @@
+package optimization
```

Comment:
i think this file might be more appropriately named `details.go`. in most cases you only see underscore for test files such as `logger_test.go` etc...

although i have seen some packages that used underscore to help split up files (and I have contemplated doing the same as well) where the package was too large. but in this case i think it's not necessary